internal/abci/abci.go

```diff
@@ -762,14 +770,16 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 	// Create a new changeset processor
 	csp := newChangesetProcessor()
 	// "migrator" module subscribes to the changeset processor to store changesets during the migration
+	csDoneChan := make(chan bool, 1)
+	csErrChan := make(chan error, 1)
 	if inMigration && !haltNetwork {
 		csChanMigrator, err := csp.Subscribe(ctx, "migrator")
 		if err != nil {
 			return nil, fmt.Errorf("failed to subscribe to changeset processor: %w", err)
 		}
 		// migrator go routine will receive changesets from the changeset processor
 		// give the new channel to the migrator to store changesets
-		go a.migrator.StoreChangesets(req.Height, csChanMigrator)
+		go a.migrator.StoreChangesets(req.Height, csChanMigrator, csDoneChan, csErrChan)
```

Comment:
I think there should just be `csErrChan`, where success in `StoreChangesets` is just closing the channel (the receiver will immediately get a `nil` `error`)

Reply:
makes sense